distribution/pull_v2.go

```diff
@@ -296,6 +300,50 @@ func (ld *v2LayerDescriptor) Download(ctx context.Context, progressOutput progre
 	}), size, nil
 }
 
+var _ distribution.Describable = &v2LayerDescriptor{}
+
+func (ld *v2LayerDescriptor) Descriptor() distribution.Descriptor {
+	if ld.src.MediaType == schema2.MediaTypeForeignLayer && len(ld.src.URLs) > 0 {
+		return ld.src
+	}
+	return distribution.Descriptor{}
+}
+
+func (ld *v2LayerDescriptor) open(ctx context.Context) (distribution.ReadSeekCloser, error) {
+	if len(ld.src.URLs) == 0 {
+		blobs := ld.repo.Blobs(ctx)
+		return blobs.Open(ctx, ld.digest)
+	}
+
+	var (
+		err error
+		rsc distribution.ReadSeekCloser
+	)
+
+	// Find the first URL that results in a 200 result code.
+	for _, u := range ld.src.URLs {
+		logrus.Debugf("Pulling %v from foreign URL %v", ld.digest, u)
+
+		purl, _ := url.Parse(u)
+		var tlsc *tls.Config
+		tlsc, err = ld.service.TLSConfig(purl.Host)
```

Comment:
Mm yeah, we need a way to get the tls config. I'll fix that when in code-review if it's not OK to have the service injected here (which I don't like either).